src/inference/src/dev/core_impl.cpp

```diff
@@ -57,8 +57,251 @@ void stripDeviceName(std::string& device, const std::string& substr) {
+        }
+        return match_type == MatchType::EXACT ? (user_value == subprop_value) : (user_value.find(subprop_value) == 0);
+        return false;
+    };
+
+    if (parsed_user_device_name._deviceName == parsed_subprop_device_name._deviceName) {
+        auto device_priority = get_device_priority_property(parsed_user_device_name._deviceName);
+        return is_matched(device_priority.prop_name, device_priority.match_type);
+    }
+
+    return false;
+}
+
+ov::Parsed ov::parseDeviceNameIntoConfig(const std::string& deviceName, const AnyMap& config) {
+    auto updated_config = config;
+    auto updated_device_name = deviceName;
+
+    /** Note: auto-batching is already applied by this time, so the call:
+     * core.compile_model("GPU", ov::device::properties("BATCH", ov::auto_batch_timeout(400)));
+     * is transformed and we have here:
+     * ov::parseDeviceNameIntoConfig("BATCH", ov::device::priorities("GPU"),
+     *                                        ov::device::properties("BATCH",
+     *                                        ov::auto_batch_timeout(400)));
+     * so, after 'flatten_sub_properties' we will have:
+     * core.compile_model("BATCH", ov::auto_batch_timeout(400),
+     *                             ov::device::priorities("GPU"));
+     *
+     * So, if one day, we want to add more options in form of ov::allow_<hetero, etc>, we need to apply it before
+     * 'flatten_sub_properties' call to have proper behavior
+     */
+
+    updated_config = flatten_sub_properties(deviceName, updated_config);
+    std::string parsed_device_priority;
+
+    // try to find ':' to extract name of virtual device
+    auto pos = deviceName.find_first_of(':');
+    if (pos != std::string::npos) {
+        updated_device_name = deviceName.substr(0, pos);
+        parsed_device_priority = deviceName.substr(pos + 1);
+    } else {
+        InferenceEngine::DeviceIDParser parser(deviceName);
+        updated_device_name = parser.getDeviceName();
+        parsed_device_priority = parser.getDeviceID();
+    }
+
+    // checks and updates device priority
+    if (!parsed_device_priority.empty()) {
+        const auto priority_prop_name = get_device_priority_property(updated_device_name).prop_name;
+        const auto it = updated_config.find(priority_prop_name);
+        if (it == updated_config.end())
+            updated_config[priority_prop_name] = parsed_device_priority;
+        else if (it->second == parsed_device_priority) {
+            // do nothing
+        } else {
+            IE_THROW() << "Device priority / ID mismatch: " << parsed_device_priority << " (from " << deviceName
+                       << ") vs " << it->second.as<std::string>() << " (from config)";
+        }
+    };
+
+    // clean-up auto-batch related properties
+    clean_batch_properties(updated_device_name, updated_config, ov::hint::allow_auto_batching.name());
+    clean_batch_properties(updated_device_name, updated_config, ov::auto_batch_timeout.name());
```

Comment:
Where to cleanup other 2 core level properties to avoid passing to pluging by compile_model() calling?

Reply:
they are cleaned up in `CoreConfig::set_and_update` method